mypy/checker.py

```diff
@@ -616,14 +608,22 @@ def is_implicit_any(t: Type) -> bool:
                 for i in range(len(typ.arg_types)):
                     arg_type = typ.arg_types[i]
 
-                    # Refuse covariant parameter type variables
-                    # TODO: check recuresively for inner type variables
-                    if isinstance(arg_type, TypeVarType):
-                        if i > 0:
-                            if arg_type.variance == COVARIANT:
-                                self.fail(messages.FUNCTION_PARAMETER_CANNOT_BE_COVARIANT,
-                                          arg_type)
-                        # FIX: if i == 0 and this is not a method then same as above
+                    ref_type = self.scope.active_class()
+                    if (isinstance(defn, FuncDef) and ref_type is not None and i == 0
```

Comment:
For me it's "Lesson 8: The State Explosion Problem" :)

This also assumes `foo` in `foo = lambda ...` is not a method (from selftype perspective), which I think is fine, but I'd expect to happen once every hundred-thousand lines or so.